xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -1184,6 +1196,42 @@ posix_unlink_gfid_handle_and_entry(call_frame_t *frame, xlator_t *this,
     return -1;
 }
 
+static int
+posix_unlink_stale_linkto(call_frame_t *frame, xlator_t *this,
+                          const char *real_path, int32_t *op_errno, loc_t *loc)
+{
+    int ret = -1;
+    struct iatt stbuf = {
+        0,
+    };
+
+    /* get stale file gfid and stat-info */
+    ret = posix_pstat(this, NULL, NULL, real_path, &stbuf, _gf_false);
+    if (ret) {
+        if (errno == ENOENT)
```

Comment:
We can only be here if the file existed. This shouldn't happen. It means we have a race somewhere. We should add a log message here.

Reply:
Added INFO logging, 
One legit race can be with rebalance.